codec-http/src/test/java/io/netty/handler/codec/http/cookie/ClientCookieEncoderTest.java

```diff
@@ -37,7 +37,9 @@ public void testEncodingMultipleClientCookies() {
         cookie2.setSecure(false);
         Cookie cookie3 = new DefaultCookie("myCookie3", "myValue3");
         String encodedCookie = ClientCookieEncoder.STRICT.encode(cookie, cookie2, cookie3);
-        assertEquals(c1 + c2 + c3, encodedCookie);
+        // Cookies should be sorted into decreasing order of path length, as per RFC6295.
+        // Where no path is provided, we assume maximum path length (so cookie3 comes first).
+        assertEquals(c3 + "; " + c2 + "; " + c1, encodedCookie);
```

Comment:
cookie1 and cookie2 do have a path -- see the `.setPath()` lines. cookie3 does not have a path. I made a judgment call in sorting path-less cookies before cookies with paths, because the sort order is supposed to be most specific to most general (longest path to shortest path, in general), and path-less cookies inherit the path of the current request.